Overhaul projects page (#217)
scala/scala.epfl.ch: projects.md

---
layout: contact
---

> 

Author: This page looks like this:

![Screenshot 2022-03-30 at 09-17-07 Scala Center at EPFL](https://user-images.githubusercontent.com/332812/160775519-d6c4ca48-60e4-46db-a5c7-0cd09011fd25.png)

This page lists the projects that the Scala Center will work on during
the current quarter. You can find more information about our completed
projects in the [quarterly activity reports](./records.html).

## Roadmap for 2022 Q2
{: .no_toc}

> The following sections present our plans for the current quarter. Every 

Reviewer: ```suggestion
The following sections present our plan for the current quarter. Every 
```

Author: Edited in 718f53d

#### Smarter “Step Into” when debugging

The “step-into” feature in Metals is hard to use because it often goes to
synthetic methods (getter, setter, bridge, mixin-forwarder), that do
not exist in the source code. We will improve the implementation of the
[scala-debug-adapter](https://github.com/scalacenter/scala-debug-adapter) to
address this issue.

#### “setup-scala” Github action

The way to set up Github Actions for the CI of Scala projects is unclear at
the moment. There used to be [olafurpg/setup-scala](https://github.com/olafurpg/setup-scala)
but that project is not maintained anymore. There are other alternatives
such as [japgolly/setup-everything-scala](https://github.com/japgolly/setup-everything-scala)
or [coursier/setup-action](https://github.com/coursier/setup-action).
Furthermore, it turns out that the official
[actions/setup-java](https://github.com/actions/setup-java) does support sbt
project. What should Scala project maintainers use? What is currently
recommended in
[the library author guide](https://docs.scala-lang.org/overviews/contributors/index.html#setup-continuous-integration)
is to use setup-java. However, this action does not support caching with

> coursier and sbt out of the box. It does not support Scala.js either. We

Reviewer: It is written above that it does support sbt. So does it?

will make sure there is a simple way to set up Github actions for typical
Scala projects, and that this action is the recommended choice in the
documentation and in the Github UI.

#### Github Security Alerts

We want to harden the security of the Scala ecosystem by implementing
Security Alerts for Scala projects using the Github API. We will publish a
Github action that can be used by library maintainers to get notified about

> vulnerabilities in their projects.

Reviewer: We only plan for sbt for now

```suggestion
vulnerabilities in their sbt projects.
```

Author: Fixed in 718f53d

address this problem by introducing a separate page for the artifacts.

Beside being an index of Scala libraries, we believe scaladex can be a 
place where Scala programmers can find open-source libraries to contribute 
to. We will implement new features on scaladex to allow library authors to 
highlight how new contributors could help them, and for new contributors to 
easily find projects to contribute to.

Last, we would like to formalize the scaladex HTTP API. Scaladex exposes a 
public HTTP API that can be used by third-party tools to search for 
libraries. However, this API is not documented, and it may change. We will 
formalize it, and document it.

#### Metals and Bloop

[Bloop](https://github.com/scalacenter/bloop) has recently been updated to use
the latest version of Zinc. However, in the process we had to remove
“pipelining”, which parallelizes better the compilation process. We will add 
back pipelining to Bloop.

Other than that, we will continue working on the remaining issues towards a 
release of [Metals 1.0.0](https://github.com/scalameta/metals/milestone/36).

#### Add Support for Binary Compatible Evolutions to Case Classes

> Case classes are a popular tool to achieve domain modeling. However, they 

Reviewer: ```suggestion
Case classes are popular to achieve domain modeling. However, they 
```

Author: Fixed in 718f53d

> come with an important drawback: it is impossible to apply evolutions 

Reviewer: ```suggestion
come with an important drawback: it is impossible to apply any evolution
```

Author: Fixed in 718f53d

(adding/removing fields on a case class) in a backward binary compatible way.

> This forces library authors to, manually re-implement the features of 

Reviewer: ```suggestion
This forces library authors to manually re-implement the features of 
```
